workflow/executor_unit_test.go

```diff
@@ -0,0 +1,436 @@
+package workflow //nolint:testpackage // Tests private members in the packages
+
+import (
+	"go.arcalot.io/assert"
+	"go.flow.arcalot.io/engine/internal/step"
+	"go.flow.arcalot.io/pluginsdk/schema"
+	"testing"
+)
+
+var testLifecycleStage = step.LifecycleStageWithSchema{
+	LifecycleStage: step.LifecycleStage{},
+	InputSchema: map[string]*schema.PropertySchema{
+		"scopeA": schema.NewPropertySchema(
+			schema.NewScopeSchema(schema.NewObjectSchema("testObjectA", map[string]*schema.PropertySchema{})),
+			nil,
+			true,
+			nil,
+			nil,
+			nil,
+			nil,
+			nil,
+		),
+		"listA": schema.NewPropertySchema(
+			schema.NewListSchema(
+				schema.NewScopeSchema(
+					schema.NewObjectSchema("testObjectB", map[string]*schema.PropertySchema{}),
+				),
+				nil,
+				nil,
+			),
+			nil,
+			true,
+			nil,
+			nil,
+			nil,
+			nil,
+			nil,
+		),
+		"intA": schema.NewPropertySchema(
+			schema.NewIntSchema(nil, nil, nil),
+			nil,
+			true,
+			nil,
+			nil,
+			nil,
+			nil,
+			nil,
+		),
+	},
+	Outputs: map[string]*schema.StepOutputSchema{
+		"outputC": schema.NewStepOutputSchema(
+			schema.NewScopeSchema(
+				schema.NewObjectSchema("testObjectC", map[string]*schema.PropertySchema{}),
+			),
+			nil,
+			false,
+		),
+	},
+}
+
+func TestAddOutputNamespacedScopes(t *testing.T) {
+	allNamespaces := make(map[string]schema.Scope)
+	expectedPrefix := "TEST_PREFIX_"
+	addOutputNamespacedScopes(allNamespaces, testLifecycleStage, expectedPrefix)
+	expectedOutput := "outputC"
+	expectedNamespace := expectedPrefix + expectedOutput
```

Comment:
At line 63, that **is** _the_ prefix -- it's not the "expected" one.  And, the local variable at line 65 seems unnecessary, given the nicely named variable constructed at line 66.  So, I recommend:
```go
	prefix := "TEST_PREFIX_"
	addOutputNamespacedScopes(allNamespaces, testLifecycleStage, prefix)
	expectedNamespace := prefix + "outputC"
```

A similar comment applies to `TestAddInputNamespacedScopes()`.